File: Museo/Clases/MuseoColaborador.py
from __future__ import annotations

import os
import sys

sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(__file__), "..")))


class MuseoColaborador:
    """Museo colaborador con el que se pueden gestionar cesiones de obras."""

    def __init__(self, nombre: str, pais: str) -> None:
        """Inicializa el museo colaborador con su nombre y país."""
        self._nombre = nombre
        self._pais = pais

    # --- Propiedades ---

    @property
    def nombre(self) -> str:
        """Retorna el nombre del museo colaborador."""
        return self._nombre

    @property
    def pais(self) -> str:
        """Retorna el país del museo colaborador."""
        return self._pais



    def __str__(self) -> str:
        return f"Museo '{self._nombre}' ({self._pais})"


# ---------------------------------------------------------------------------
# Prueba
# ---------------------------------------------------------------------------

if __name__ == "__main__":
    print("=== Prueba de MuseoColaborador ===")

    nombre = input("Nombre del museo colaborador: ")
    pais = input("País: ")

    museo = MuseoColaborador(nombre, pais)
    print(f"\n{museo}")

File: 2.2Museo_Java/Clases_Museo/MuseoColaborador.java
package Clases_Museo;

/**
 * Museo colaborador con el que se pueden gestionar cesiones de obras.
 *
 * Almacena el nombre y el pais del museo para identificarlo
 * dentro del sistema de cesiones.
 */
public class MuseoColaborador {

    // Atributos privados
    private String nombre;
    private String pais;

    // --- Constructor ---

    /**
     * Inicializa el museo colaborador con su nombre y pais.
     *
     * @param nombre Nombre del museo colaborador.
     * @param pais   Pais donde se ubica el museo.
     */
    public MuseoColaborador(String nombre, String pais) {
        this.nombre = nombre;
        this.pais   = pais;
    }

    // --- Getters ---

    /** @return El nombre del museo colaborador. */
    public String getNombre() { return nombre; }

    /** @return El pais donde se ubica el museo colaborador. */
    public String getPais() { return pais; }

    @Override
    public String toString() {
        return "Museo '" + nombre + "' (" + pais + ")";
    }
}
